Changed deliverable to milestone

kelleyb opened the pull request:
Apparently Richie forgot.

kelleyb committed:
Changed deliverable to milestone

kmcnellis commented:
I think the term "milestone" is a bit of a different connotation, and therefore needs explaining on what we expect a milestone to be.  For instance, in github, a alpha or beta release could be considered a milestone, but we want discrete sets of features (not just a project getting to a certain point)

kmcnellis commented:
Maybe add something like "Each milestone should be a overarching goal, consisting of a well-defined large-scale feature, and be able to be broken up into several issues."

kelleyb commented:
Yeah, I'll add a bit explaining that.

kelleyb committed:
Clarified milestones

Described milestones a little more, explained the difference in general connotation used between GitHub and RCOS.

aeksco commented:
@kelleyb What's the status of this PR? I think the changes here are solid - we'll likely re-tool them a bit (as per @kmcnellis 's comment)  during the summer as we prepare for the Fall 2018 semester. LMK your thoughts when you have a chance - thanks!
